### ISSUE TITLE
server blocks added to nginx.conf

### ISSUE THREAD

**Issue by jsha** [certbot/certbot#925]
Steps to reproduce:

(on Ubuntu)
```
letsencrypt --server SERVER --installer nginx --authenticator nginx --text --agree-eula --agree-tos --no-verify-ssl -d 4.le-test.hoffman-andrews.com run
```

Expected result:

A new file named 4.le-test.hoffman-andrews.com is created under /etc/nginx/sites-available and symlinked into /etc/nginx/sites-enabled/.

Actual result:

A server block is added to /etc/nginx/nginx.conf:

```
    server {
        server_name 4.le-test.hoffman-andrews.com;
        listen 443 ssl;
        access_log /var/lib/letsencrypt/access.log;
        error_log /var/lib/letsencrypt/error.log;
        ssl_certificate /etc/letsencrypt/live/4.le-test.hoffman-andrews.com/cert.pem;
        ssl_certificate_key /etc/letsencrypt/live/4.le-test.hoffman-andrews.com/privkey.pem;
        include /etc/letsencrypt/options-ssl-nginx.conf;
    }
```

I know this is probably a hard problem to solve nicely, because they layout of where each virtualhost goes is a distribution-dependent thing. If we don't want to have special logic for the distribution's layouts of virtualhosts, maybe we could just default to putting them in a separate file that gets included in, e.g /etc/nginx/letsencrypt-virtualhosts.conf, or /etc/letsencrypt/nginx-virtualhosts.conf.

**Comment by tdruiva** [comment]
Based on the conversation in this PR #2097 , I'll going to create a PR  to remove all vhost creation.
@joohoi @pde Thoughts?

**Comment by jsha** [comment]
Makes sense to me.

**Comment by pde** [comment]
Over in Apache land, we've gradually developed a [collection of machinery](https://github.com/certbot/certbot/blob/master/certbot-apache/certbot_apache/constants.py) for knowing how each host OS will want to store its vhosts/server blocks. That's probably the right way to do things, if we are going to keep making new server blocks in the nginx plugin (in at least some cases).

**Comment by bmw** [comment]
Fixed in #3528.
